wp-content/plugins/wporg-learn/inc/post-meta.php

```diff
@@ -257,32 +257,32 @@ function get_workshop_duration( WP_Post $workshop, $format = 'raw' ) {
 }
 
 /**
- * Get a list of locales that are associated with at least one workshop.
- *
- * Optionally only published workshops.
+ * Get a list of locales that are associated with at least one post of the specified type.
  *
  * @param string $meta_key
+ * @param string $post_type
  * @param string $label_language
- * @param bool   $published_only
+ * @param string $post_status
  *
  * @return array
  */
-function get_available_workshop_locales( $meta_key, $label_language = 'english', $published_only = true ) {
+function get_available_post_type_locales( $meta_key, $post_type, $label_language = 'english', $post_status ) {
 	global $wpdb;
 
 	$and_post_status = '';
-	if ( $published_only ) {
-		$and_post_status = "AND posts.post_status = 'publish'";
+	if ( in_array( $post_status, get_post_stati(), true ) ) {
```

Comment:
@iandunn are you satisfied with the security of this approach? `$post_status` is passed in from `INPUT_GET` [here](https://github.com/WordPress/Learn/pull/1500/files#diff-2644427c2043c94e7dcc0a36501c5142deb7b72c046baa6210e2cd61cf16b8a3R238).

This line attempts to ensure that only safe strings matching `post_status` values in the system are included in the query.

Reply:
This is admin only too. On the frontend `post_status` is always hard coded to `publish`.